addons/payment_stripe/models/payment_transaction.py

```diff
@@ -339,12 +415,16 @@ def _process_notification_data(self, notification_data):
         if self.provider != 'stripe':
             return
 
-        if 'charge' in notification_data:
+        if self.operation == 'refund':
+            self.acquirer_reference = notification_data['refund']['id']
+        elif 'charge' in notification_data:
```

Comment:
As discussed, It would be better indeed to check the operation only or the keys but the `refunds` key may be there with no refunds and the `charge` key may not be there for other transactions. For this very specific case, I would advise keeping this code like it, but I'll add the comment that you suggested :)